src/resources/grammar/__tests__/SearchDSL-test.js

```diff
@@ -0,0 +1,158 @@
+jest.dontMock('../../../js/utils/DSLParserUtil');
+
+const SearchDSL = require('../SearchDSL.jison');
+const DSLFilterTypes = require('../../../js/constants/DSLFilterTypes');
+const DSLCombinerTypes = require('../../../js/constants/DSLCombinerTypes');
+
+describe('SearchDSL', function () {
+
+  describe('Metadata', function () {
+
+    describe('Filters (Operands)', function () {
+
+      it('should parse fuzzy text', function () {
+        let expr = SearchDSL.parse('fuzzy');
+        expect(expr.ast.filterType).toEqual(DSLFilterTypes.FUZZY);
+        expect(expr.ast.filterParams.text).toEqual('fuzzy');
+      });
+      it('should parse exact text', function () {
+        let expr = SearchDSL.parse('"exact string"');
+        expect(expr.ast.filterType).toEqual(DSLFilterTypes.EXACT);
+        expect(expr.ast.filterParams.text).toEqual('exact string');
+      });
+      it('should parse attributes', function () {
+        let expr = SearchDSL.parse('attrib:value');
+        expect(expr.ast.filterType).toEqual(DSLFilterTypes.ATTRIB);
+        expect(expr.ast.filterParams.text).toEqual('value');
+        expect(expr.ast.filterParams.label).toEqual('attrib');
+      });
+
+    });
+
+    describe('Combiners (Operators)', function () {
+
+      it('should use AND operator by default', function () {
+        let expr = SearchDSL.parse('text1 text2');
+        expect(expr.ast.combinerType).toEqual(DSLCombinerTypes.AND);
+      });
+
+      it('should properly use OR operator', function () {
+        let expr = SearchDSL.parse('text1, text2');
+        expect(expr.ast.combinerType).toEqual(DSLCombinerTypes.OR);
+      });
+
+      it('should properly use OR shorthand operator on attrib', function () {
+        // NOTE: attrib:value1,value2 becomes -> attrib:value1, attrib:value2
+        let expr = SearchDSL.parse('attrib:value1,value2');
+        expect(expr.ast.combinerType).toEqual(DSLCombinerTypes.OR);
+        expect(expr.ast.children[0].filterType).toEqual(DSLFilterTypes.ATTRIB);
+        expect(expr.ast.children[0].filterParams.text).toEqual('value1');
+        expect(expr.ast.children[0].filterParams.label).toEqual('attrib');
+        expect(expr.ast.children[1].filterType).toEqual(DSLFilterTypes.ATTRIB);
+        expect(expr.ast.children[1].filterParams.text).toEqual('value2');
+        expect(expr.ast.children[1].filterParams.label).toEqual('attrib');
+      });
+
+      it('should correctly populate children', function () {
+        let expr = SearchDSL.parse('text1 text2');
+        expect(expr.ast.children[0].filterType).toEqual(DSLFilterTypes.FUZZY);
+        expect(expr.ast.children[0].filterParams.text).toEqual('text1');
+        expect(expr.ast.children[1].filterType).toEqual(DSLFilterTypes.FUZZY);
+        expect(expr.ast.children[1].filterParams.text).toEqual('text2');
+      });
+
+    });
+
+    describe('Complex expressions', function () {
+
+      it('should nest operations in correct order', function () {
```

Comment:
We could add more tests for each individual case (in addition), but I would vote for keeping this one, because I am also testing the order of the operations.